File: src/main/java/com/jocoos/mybeautip/domain/system/persistence/domain/SystemOption.java
package com.jocoos.mybeautip.domain.system.persistence.domain;

import com.jocoos.mybeautip.domain.system.code.SystemOptionType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Entity
public class SystemOption {

    @Id
    @Enumerated(EnumType.STRING)
    private SystemOptionType id;

    @Column
    private boolean value;

    public void updateValue(boolean value) {
        this.value = value;
    }
}

File: src/test/java/com/jocoos/mybeautip/testutil/fixture/SystemOption.kt
package com.jocoos.mybeautip.testutil.fixture

import com.jocoos.mybeautip.domain.system.code.SystemOptionType
import com.jocoos.mybeautip.domain.system.persistence.domain.SystemOption

fun makeSystemOption(
    id: SystemOptionType = SystemOptionType.FREE_LIVE_PERMISSION,
    value: Boolean = false
): SystemOption {
    return SystemOption(
        id,
        value
    )
}